django/db/models/lookups.py

```diff
@@ -224,13 +224,13 @@ def process_rhs(self, compiler, connection):
     def get_rhs_op(self, connection, rhs):
         return 'IN %s' % rhs
 
-    def as_sql(self, compiler, connection):
+    def as_sql(self, compiler, connection, **kwargs):
         max_in_list_size = connection.ops.max_in_list_size()
         if self.rhs_is_direct_value() and max_in_list_size and len(self.rhs) > max_in_list_size:
-            return self.split_parameter_list_as_sql(compiler, connection)
-        return super(In, self).as_sql(compiler, connection)
+            return self.split_parameter_list_as_sql(compiler, connection, **kwargs)
```

Comment:
I'm not sure that this particular call requires `**kwargs`. Check the actual function definition to see if it will require kwargs or not. I don't think it is overridable with `as_vendorname`.